backend/createStaffAndRooms.py

```diff
@@ -44,43 +45,104 @@ def search_ticket(ticket, guest_entries):
     return False
 
 
-def create_rooms_main(rooms_file ="", is_hardrock=False):
+def real_date(a_date):
+    day, date = a_date.split('-')
+    month, day = date.lstrip().split('/')
+    return parse_date("%s-%s-%s" % (datetime.now().year, month, day))
+
+def create_rooms_main(rooms_file, is_hardrock=False):
     rooms=[]
-    rooms_rows = []
-    with open(rooms_file, "r") as rfile:
-        for row in DictReader(rfile):
-            stripd = {k.lstrip().rstrip(): v.lstrip().rstrip() for k, v in row.items() if type(k)==str and type(v)==str}
-            rooms_rows.append(stripd)
+    _rooms_fields, rooms_rows = ingest_csv(rooms_file)
+
+    if(is_hardrock):
+        hotel = "Hard Rock"
+    else:
+        hotel = "Ballys"
+
+    logger.info("read in %s rooms for %s", len(rooms_rows), hotel)
 
-         
     for elem in rooms_rows:
-        if(elem["Placed By"]=="Roombaht"):
-            if(is_hardrock):
-                hotel = "Hard Rock"
+        a_room = Room(name_take3=elem['Room Type'],
+                      name_hotel=hotel,
+                      number=elem['Room']
+                      )
+
+        # these things are always going to be consistent per room
+        if elem['Check-in Date'] != '':
+            a_room.check_in = real_date(elem['Check-in Date'])
+
+        if elem['Check-out Date'] != '':
+            a_room.check_out = real_date(elem['Check-out Date'])
+
+        features = elem['Room Features (Accesbility, Lakeview, Smoking)(not visible externally)'].lower()
+        if 'hearing accessible' in features:
+            a_room.is_hearing_accessible = True
+
+        if 'ada' in features:
+            a_room.is_ada = True
+
+        if 'lakeview' in features:
+            a_room.is_lakeview = True
+
+        if 'smoking' in features:
+            a_room.is_smoking = True
+
+        if len(elem['Room Notes']) > 0:
+            a_room.notes = elem['Room Notes']
+
+        if elem['Changeable'] == '' or \
+           'yes' in elem['Changeable'].lower():
+            a_room.is_swappable = True
+
+        if elem['Placed By'] == 'Roombaht':
+            a_room.is_available = True
```

Comment:
nit for future:
All of these are hard-coding the column names from the spreadsheet. At some point it may be useful to have a map definition from the venue spreadsheet columns into internally defined & static column names. So if different venues (or parties) have different room allocation sheets, we can have a config that maps it into our schema.

Don't worry about it here fo sho. Just wanted to name it.

Reply:
Agree :100: - we've made some progress with this kinda stuff and the bits that are now in `constants.py`.

For a future (like, next year future) iteration we should move all these maps out into yaml files or something.